packages/plugins/src/plugins/pendle/helpers.ts

```diff
@@ -0,0 +1,197 @@
+import {
+  EvmNetworkIdType,
+  networks,
+  PortfolioElementLabel,
+  TokenPriceSource,
+  UniTokenInfo,
+} from '@sonarwatch/portfolio-core';
+import axios from 'axios';
+import { Address, ContractFunctionConfig, getAddress } from 'viem';
+import {
+  BASE_API_URL,
+  MERKLE_TREE_GITHUB_REPO_URL,
+  platformId,
+} from './constants';
+import {
+  ApiResponse,
+  BaseAssetType,
+  GithubContent,
+  MerkleTreeJSON,
+  PendleAsset,
+  PendleResult,
+} from './types';
+import { getEvmClient } from '../../utils/clients';
+import { lpAbi, syAbi, ytAbi } from './abi';
+import { zeroBigInt } from '../../utils/misc/constants';
+
+export async function getMarkets(
+  networkId: EvmNetworkIdType
+): Promise<PendleResult[]> {
+  const limit = 100;
+  let skip = 0;
+  let total = 0;
+  const allResults: PendleResult[] = [];
+
+  do {
+    const { data } = await axios.get<ApiResponse>(
+      `${BASE_API_URL}/${networks[networkId].chainId}/markets`,
+      { params: { limit, skip } }
+    );
+
+    if (total === 0 && data.total) total = data.total;
+
+    if (!data.results || data.results.length === 0) break;
+
+    allResults.push(...data.results);
+    skip += data.results.length;
+  } while (skip < total);
+
+  return allResults;
+}
+
+export function formatPendleAssetsToPriceSources(
+  assets: PendleAsset[],
+  networkId: EvmNetworkIdType
+): TokenPriceSource[] {
+  const timestamp = Date.now();
+  return assets.map((asset) => ({
+    id: asset.id,
+    weight: 0.7,
+    address: getAddress(asset.address),
+    networkId,
+    platformId,
+    decimals: asset.decimals,
+    price: asset.price.usd,
+    elementName: asset.name,
+    timestamp,
+  }));
+}
+
+export function buildTokenMetaDataItems(
+  tokens: PendleAsset[],
+  networkId: EvmNetworkIdType
+): UniTokenInfo[] {
+  return tokens.map((token) => ({
+    chainId: networks[networkId].chainId,
+    address: getAddress(token.address),
+    decimals: token.decimals,
+    name: token.simpleName || token.name,
+    symbol: token.symbol,
+    logoURI: token.simpleIcon,
+    extensions: {},
+  }));
+}
+
+export function mapAssetTypeToLabel(
+  assetType: BaseAssetType
+): PortfolioElementLabel {
+  if (assetType === 'PT' || assetType === 'YT') return 'Deposit';
+  if (assetType === 'SY') return 'Staked';
+  if (assetType === 'PENDLE_LP') return 'LiquidityPool';
+
+  throw new Error(`Unsupported asset type: ${assetType}`);
+}
+
+function generateYtContractConfig(
```

Comment:
```suggestion
function generateYTContractConfig(
```

Use upper-cased letter for the 2nd one like the other methods ?